graphql_compiler/interpreter/typedefs.py

```diff
@@ -272,5 +272,103 @@ def can_coerce_to_type(
+                    current_token = data_context.current_token
+                    can_coerce: bool
+                    if current_token is None:
+                        # Evaluating an @optional scope where the optional edge didn't exist.
+                        # We cannot coerce something that doesn't exist.
+                        can_coerce = False
+                    else:
+                        can_coerce = (
+                            < check whether current_token represents a vertex
+                                of type coerce_to_type_name >
+                        )
+
+                    # Remember to always yield the DataContext alongside the bool result.
+                    yield data_context, can_coerce
+
+        Args:
+            data_contexts: iterable of DataContext objects which specify the DataTokens that are
+                           being coerced to a new type
+            current_type_name: name of the vertex type from which the vertices are being coerced.
+                               Guaranteed to be the name of an interface or union type defined
+                               in the schema being queried.
+            coerce_to_type_name: name of the vertex type to which the vertices are being coerced.
+                                 Guaranteed to be the name of a type defined in the schema being
+                                 queried. If current_type_name refers to an interface type, then
+                                 coerce_to_type_name is guaranteed to refer to a type that
+                                 implements that interface. If current_type_name refers to a union
+                                 type, then coerce_to_type_name is guaranteed to refer to a type
+                                 that is a member of that union type.
+            runtime_arg_hints: names and values of any runtime arguments provided to the query
+                               for use in filtering operations (e.g. "$arg_name").
+            used_property_hints: the property names of the vertices being processed that
+                                 are going to be used in a subsequent filtering or output step.
+            filter_hints: information about any filters applied to the vertices being processed,
+                          such as "which filtering operations are being performed?"
+                          and "with which arguments?"
+            neighbor_hints: information about the edges of the vertices being processed
+                            that the query will eventually need to expand.
+            **hints: catch-all kwarg field making the function's signature forward-compatible with
+                     future revisions of this library that add more hints.
+
+        Yields:
+            tuples (data_context, can_coerce), with a DataContext together with whether
+            the current_token referenced by that context can be coerced to the specified type.
```

Comment:
```suggestion
            tuples (data_context, can_coerce), containing a DataContext and a corresponding boolean indicating
            whether its current_token can be coerced to the specified type.
```
nit: the `with`/`with` got me a little tripped up.

Reply:
I like this! One tweak I'd suggest is working around that `its` in `its current_token` since it is not obvious that it refers to `DataContext` and not `corresponding boolean`. Maybe if I change it to `indicating whether the DataContext's current_token` instead? Let me know what you think.